tree/tree.go

```diff
@@ -0,0 +1,651 @@
+	PageDown     key.Binding
+	PageUp       key.Binding
+	HalfPageUp   key.Binding
+	HalfPageDown key.Binding
+	GoToTop      key.Binding
+	GoToBottom   key.Binding
+
+	Toggle key.Binding
+	Open   key.Binding
+	Close  key.Binding
+
+	// Help toggle keybindings.
+	ShowFullHelp  key.Binding
+	CloseFullHelp key.Binding
+
+	Quit key.Binding
+}
+
+// DefaultKeyMap returns the default set of key bindings for navigating and acting
+// upon the tree.
+func DefaultKeyMap() KeyMap {
+	return KeyMap{
+		Down: key.NewBinding(
+			key.WithKeys("down", "j", "ctrl+n"),
+			key.WithHelp("↓/j", "down"),
+		),
+		Up: key.NewBinding(
+			key.WithKeys("up", "k", "ctrl+p"),
+			key.WithHelp("↑/k", "up"),
+		),
+		PageDown: key.NewBinding(
+			key.WithKeys("pgdown", spacebar, "f"),
+			key.WithHelp("f/pgdn", "page down"),
+		),
+		PageUp: key.NewBinding(
+			key.WithKeys("pgup", "b"),
+			key.WithHelp("b/pgup", "page up"),
+		),
+		HalfPageDown: key.NewBinding(
+			key.WithKeys("d", "ctrl+d"),
+			key.WithHelp("d", "½ page down"),
+		),
+		HalfPageUp: key.NewBinding(
+			key.WithKeys("u", "ctrl+u"),
+			key.WithHelp("u", "½ page up"),
+		),
+		GoToTop: key.NewBinding(
+			key.WithKeys("g", "home"),
+			key.WithHelp("g", "top"),
+		),
+		GoToBottom: key.NewBinding(
+			key.WithKeys("G", "shift+g", "end"),
+			key.WithHelp("G", "bottom"),
+		),
+
+		Toggle: key.NewBinding(
+			key.WithKeys("enter"),
+			key.WithHelp("⏎", "toggle"),
+		),
+		Open: key.NewBinding(
+			key.WithKeys("l", "right"),
+			key.WithHelp("→/l", "open"),
+		),
+		Close: key.NewBinding(
+			key.WithKeys("h", "left"),
+			key.WithHelp("←/h", "close"),
+		),
+
+		// Toggle help.
+		ShowFullHelp: key.NewBinding(
+			key.WithKeys("?"),
+			key.WithHelp("?", "more"),
+		),
+		CloseFullHelp: key.NewBinding(
+			key.WithKeys("?"),
+			key.WithHelp("?", "close help"),
+		),
+
+		Quit: key.NewBinding(
+			key.WithKeys("q", "ctrl+c"),
+			key.WithHelp("q", "quit"),
+		),
+	}
+}
+
+// Model is the Bubble Tea model for this tree element.
+type Model struct {
+	showHelp bool
+	// scrollOff is the minimal number of lines to keep visible above and below the selected node.
+	scrollOff int
+	// openCharacter is the character used to represent an open node.
+	openCharacter string
+	// closedCharacter is the character used to represent a closed node.
+	closedCharacter string
+	// cursorCharacter is the character used to represent the cursor.
+	cursorCharacter string
+	// KeyMap encodes the keybindings recognized by the widget.
+	KeyMap KeyMap
+	// styles sets the styling for the tree
+	styles Styles
+	Help   help.Model
+
+	// Additional key mappings for the short and full help views. This allows
+	// you to add additional key mappings to the help menu without
+	// re-implementing the help component. Of course, you can also disable the
+	// list's help component and implement a new one if you need more
+	// flexibility.
+	AdditionalShortHelpKeys func() []key.Binding
+	AdditionalFullHelpKeys  func() []key.Binding
```

Comment:
I feel that these two should also be private with `Set*` functions.

Reply:
I've changed these, but just noted that these are public fields in the list component.
Let me know if you want me to revert / change the list as well.

Reply:
I still prefer to keep them private. It may have been by mistake that these got public on the other package.